sda/internal/config/config.go

```diff
@@ -1066,6 +1071,35 @@ func GetC4GHKey() (*[32]byte, error) {
 	return &key, nil
 }
 
+// GetC4GHprivateKeys reads and decrypts keys and returns a list of c4gh keys
+func GetC4GHprivateKeys() ([]*[32]byte, error) {
+	// Retrieve the list of key configurations from the YAML file
+	var keyConfigs []C4GHprivateKeyConf
+	if err := viper.UnmarshalKey("c4gh.privateKeys", &keyConfigs); err != nil {
+		return nil, fmt.Errorf("failed to parse key configurations: %v", err)
+	}
+
+	var privateKeys []*[32]byte
+
+	// Iterate over each key configuration
+	for _, keyConfig := range keyConfigs {
+		keyFile, err := os.Open(keyConfig.FilePath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open key file %s: %v", keyConfig.FilePath, err)
+		}
+
+		key, err := keys.ReadPrivateKey(keyFile, []byte(keyConfig.Passphrase))
+		keyFile.Close()
+		if err != nil {
+			return nil, fmt.Errorf("failed to read private key from %s: %v", keyConfig.FilePath, err)
+		}
+
+		privateKeys = append(privateKeys, &key)
+	}
+
+	return privateKeys, nil
+}
```

Comment:
Still not happy with the existence of `conf` and `Config` in this block, `keySet` or similar is a better name.

Also this function needs its own test case in `config_test.go` in order to test all the ways it can break.

Reply:
Fixed in 7921d7eb